src/distPlus/distribution/Discrete.re

```diff
@@ -87,68 +100,80 @@ let combineAlgebraically =
 
   let combinedShape = XYShape.T.fromZippedArray(rxys);
 
-  make(combinedShape, combinedIntegralSum);
+  make(combinedShape, combinedIntegralSum, None);
 };
 
-let mapY = (~knownIntegralSumFn=previousKnownIntegralSum => None, fn, t: t) => {
-  let u = E.O.bind(_, knownIntegralSumFn);
+let mapY = (~integralSumCacheFn=_ => None,
+            ~integralCacheFn=_ => None,
+            fn, t: t) => {
   let yMapFn = shapeMap(XYShape.T.mapY(fn));
 
-  t |> yMapFn |> updateKnownIntegralSum(u(t.knownIntegralSum));
+  t
+  |> yMapFn
+  |> updateIntegralSumCache(E.O.bind(t.integralSumCache, integralSumCacheFn))
+  |> updateIntegralCache(E.O.bind(t.integralCache, integralCacheFn));
 };
 
+
 let scaleBy = (~scale=1.0, t: t): t => {
+  let scaledIntegralSumCache = E.O.bind(t.integralSumCache, v => Some(scale *. v));
+  let scaledIntegralCache = E.O.bind(t.integralCache, v => Some(Continuous.scaleBy(~scale, v)));
+
   t
   |> mapY((r: float) => r *. scale)
-  |> updateKnownIntegralSum(
-       E.O.bind(t.knownIntegralSum, v => Some(scale *. v)),
-     );
+  |> updateIntegralSumCache(scaledIntegralSumCache)
+  |> updateIntegralCache(scaledIntegralCache)
 };
 
 module T =
   Dist({
     type t = DistTypes.discreteShape;
     type integral = DistTypes.continuousShape;
-    let integral = (~cache, t) =>
-      if (t |> getShape |> XYShape.T.length > 0) {
-        switch (cache) {
-        | Some(c) => c
-        | None =>
-          Continuous.make(
-            `Stepwise,
-            XYShape.T.accumulateYs((+.), getShape(t)),
-            None,
-          )
-        };
-      } else {
+    let integral = (t) =>
+      if (t |> getShape |> XYShape.T.isEmpty) {
         Continuous.make(
           `Stepwise,
           {xs: [|neg_infinity|], ys: [|0.0|]},
           None,
+          None,
         );
+      } else {
+        switch (t.integralCache) {
+        | Some(c) => c
+        | None => {
+            let ts = getShape(t);
+            // The first xy of this integral should always be the zero, to ensure nice plotting
+            let firstX = ts |> XYShape.T.minX;
+            let prependedZeroPoint: XYShape.T.t = {xs: [|firstX -. epsilon_float|], ys: [|0.|]};
```

Comment:
I could see having a float function for ``minusEpsilon``. It's kind of weird to have a function for a specific value, but this one is kind of weird, so I could see it.

Reply:
It could make sense to have a new type called "point", which is like,
```
type xyPoint = {x: float, y: float}
```

Then use that in places like this.